src/modules/rtpengine/rtpengine.c

```diff
@@ -3376,10 +3426,12 @@ static int parse_flags(struct ng_flags_parse *ng_flags, struct sip_msg *msg,
 				} else if(str_eq(&key, "AVP") && !val.s) {
 					ng_flags->transport |= 0x100;
 					ng_flags->transport &= ~0x002;
-				} else if(str_eq(&key, "TOS") && val.s)
+				} else if(str_eq(&key, "TOS") && val.s) {
 					bencode_dictionary_add_integer(
 							ng_flags->dict, "TOS", atoi(val.s));
-				else
+				} else if(str_eq(&key, "all")) {
+					ng_flags->all_legs = 1;
+				} else
```

Comment:
I think this is redundant? Just putting the word "all" in the argument string would already lead to "all" being put in the flags list without the need for special handling, no?

Reply:
I was hoping so too,
unknown flags get put into ng_flags->flags

however
https://github.com/kamailio/kamailio/blob/7de2b9b7625d9cb856944627a3a5d7c167006dc6/src/modules/rtpengine/rtpengine.c#L3594

ignores ng_flags->flags when doing the encoding.  I was not brave enough to change this behaviour as the effect would be quite global!

Reply:
I don't quite follow. The flags are constructed within `rtpp_function_call` and then put into the dictionary: https://github.com/kamailio/kamailio/blob/7de2b9b7625d9cb856944627a3a5d7c167006dc6/src/modules/rtpengine/rtpengine.c#L3730

Reply:
I've re-tested with the "all" flag removed and it still gets put in flags.  So the only thing lost is enforcing that either "all" or "from_tags" is sent, but not both. updating the patch to remove this code